From a9bdf904a0db776d47b9ff21a0be5237af28b9c3 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Fri, 22 Jan 2021 15:21:42 +0800
Subject: [PATCH] PCI: rockchip: Disable ltssm after waiting for L2 handshake

We should not disable it before L2, otherwise all L2 transaction
event would be lost for RC side, leading to unstable L2 link stage.

Change-Id: Ia87f8749d4a0acc9aeb1215682209c853672a84d
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/pci/controller/pcie-rockchip-host.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/pci/controller/pcie-rockchip-host.c b/drivers/pci/controller/pcie-rockchip-host.c
index a0ec98c05c96..45547ee5e2cd 100644
--- a/drivers/pci/controller/pcie-rockchip-host.c
+++ b/drivers/pci/controller/pcie-rockchip-host.c
@@ -981,10 +981,6 @@ static int rockchip_pcie_suspend_for_user(struct device *dev)
 	struct rockchip_pcie *rockchip = dev_get_drvdata(dev);
 	int ret;
 
-	/* disable ltssm */
-	rockchip_pcie_write(rockchip, PCIE_CLIENT_LINK_TRAIN_DISABLE,
-			    PCIE_CLIENT_CONFIG);
-
 	/* disable core and cli int since we don't need to ack PME_ACK */
 	rockchip_pcie_write(rockchip, (PCIE_CLIENT_INT_CLI << 16) |
 			    PCIE_CLIENT_INT_CLI, PCIE_CLIENT_INT_MASK);
@@ -996,6 +992,10 @@ static int rockchip_pcie_suspend_for_user(struct device *dev)
 		return ret;
 	}
 
+	/* disable ltssm */
+	rockchip_pcie_write(rockchip, PCIE_CLIENT_LINK_TRAIN_DISABLE,
+			    PCIE_CLIENT_CONFIG);
+
 	rockchip_pcie_deinit_phys(rockchip);
 
 	return ret;
-- 
2.35.3

